lib/httplib/httplib_test.go

```diff
@@ -407,3 +407,63 @@ func TestSetRedirectPageContentSecurityPolicy(t *testing.T) {
 		require.Contains(t, actualCsp, expectedCspSubString)
 	}
 }
+
+func TestOriginLocalRedirectURI(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name     string
+		input    string
+		expected string
+		wantErr  bool
```

Comment:
```suggestion
		check  require.ErrorAssertionFunc
```

And then set `check: require.NoError` or `check: require.Error` in each test case.

This allows you to remove the if statement in the test body and just say `tc.check(t, err)`

Reply:
Good suggestion, updated